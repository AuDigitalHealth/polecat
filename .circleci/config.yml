defaults: &defaults
  working_directory: &default_workdir ~/repo
  docker:
    - image: circleci/node:8.11.3

version: 2
jobs:
  build:
    <<: *defaults
    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-
      - run: yarn install
      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}
      - run: yarn lint
      - run: yarn test-coverage
      - run: yarn codecov
      - run: yarn build
      - persist_to_workspace:
          root: .
          paths:
            - build
  deploy:
    <<: *defaults
    steps:
      - checkout
      - attach_workspace:
          at: *default_workdir
      - add_ssh_keys
      - setup_remote_docker
      - run: yarn dockerize
      - run: yarn push-docker
      - run: ssh-keyscan $SERVER_HOSTNAME >> ~/.ssh/known_hosts
      - run: yarn deploy
      - run: yarn record-release

workflows:
  version: 2
  build_and_deploy:
    jobs:
      - build
      - deploy:
          requires:
            - build
          filters:
            branches:
              only: master
