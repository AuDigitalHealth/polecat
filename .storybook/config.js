import { configure, addDecorator } from '@storybook/react'
import StoryRouter from 'storybook-router'
import { withKnobsOptions } from '@storybook/addon-knobs'
import { withSmartKnobs } from 'storybook-addon-smart-knobs'

addDecorator(withSmartKnobs)
addDecorator(withKnobsOptions({ debounce: { wait: 350 } }))
addDecorator(StoryRouter())

const req = require.context('../src', true, /\.stories\.js$/)

function loadStories() {
  req.keys().forEach(filename => req(filename))
}

configure(loadStories, module)
