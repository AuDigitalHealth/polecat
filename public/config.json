{
  "fhirServer": "http://medserve.local/fhir"
}
