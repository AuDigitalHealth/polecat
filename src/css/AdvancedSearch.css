.search-advanced {
  padding: 0;
  display: flex;
  overflow: hidden;
}

.search-advanced-form {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 20px;
  height: 100%;
  position: relative;
  box-sizing: border-box;
}

.search-advanced-form .search-toggle-advanced {
  position: absolute;
  top: 29px;
  right: 28px;
}

.search-advanced .search-advanced-form > .loading img {
  position: absolute;
  top: 28px;
  right: 27px;
  width: 17px;
  height: 17px;
}

.search-advanced-results {
  background-color: rgba(255, 255, 255, 0.9);
  border-left: 1px solid #ccc;
  min-width: 0;
  flex-grow: 1;
  padding-top: 12px;
}

.search-advanced-results .search-summary {
  padding: 0 20px 10px;
  border-bottom: 1px solid #ccc;
}

.search-advanced-form .loading-or-expand.loading .search-toggle-advanced {
  display: none;
}
