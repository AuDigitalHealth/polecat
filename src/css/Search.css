.search {
  font-family: 'Helvetica Neue', sans-serif;
  position: absolute;
  top: 0;
  left: 0;
  padding: 20px;
}

.search-input {
  font-family: 'Helvetica Neue', sans-serif;
  border: 1px solid #666;
  border-radius: 6px;
  padding: 7px 27px 7px 7px;
  font-size: 14px;
  width: 229px;
}

.search-input:disabled {
  background-color: #eee;
}

.search-toggle-advanced {
  position: absolute;
  top: 23px;
  right: 29px;
}

.search .search-basic-form {
  position: relative;
  width: 266px;
}

.search .search-basic-form .search-toggle-advanced {
  top: 3px;
  right: 10px;
}

.search-basic-form .loading-or-expand.loading .search-toggle-advanced,
.search-advanced-form .loading-or-expand.loading .search-toggle-advanced {
  display: none;
}

.search .search-basic-form .loading img {
  top: 8px;
  right: 8px;
  width: 17px;
  height: 17px;
}

.search .search-advanced-form .loading img {
  top: 28px;
  right: 27px;
  width: 17px;
  height: 17px;
}

.search.search-advanced {
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  padding: 0;
  display: flex;
}

.search .search-advanced-form {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 20px;
  height: 100%;
  position: relative;
}

.search .search-advanced-results {
  background-color: rgba(255, 255, 255, 0.9);
  border-left: 1px solid #ccc;
  min-width: 0;
  flex-grow: 1;
  padding-top: 12px;
}

.search-advanced-results .search-summary {
  padding: 0 20px 10px;
  border-bottom: 1px solid #ccc;
}
