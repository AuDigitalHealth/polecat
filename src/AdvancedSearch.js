import React, { Component } from 'react'
import PropTypes from 'prop-types'

import TextField from './TextField.js'
import SearchForm from './SearchForm.js'
import Loading from './Loading.js'
import Expand from './Expand.js'
import SearchSummary from './SearchSummary.js'
import FullSearchResults from './FullSearchResults.js'
import { nextLinkFromBundle, previousLinkFromBundle } from './fhir/bundle.js'

import './css/AdvancedSearch.css'

class AdvancedSearch extends Component {
  static propTypes = {
    fhirServer: PropTypes.string,
    routedQuery: PropTypes.string,
    currentQuery: PropTypes.string,
    results: PropTypes.array,
    bundle: PropTypes.shape({ total: PropTypes.number }),
    loading: PropTypes.bool,
    onQueryUpdate: PropTypes.func,
    onToggleAdvanced: PropTypes.func,
    onNextClick: PropTypes.func,
    onPreviousClick: PropTypes.func,
    onSelectResult: PropTypes.func,
  }

  constructor(props) {
    super(props)
    this.handleQueryUpdate = this.handleQueryUpdate.bind(this)
    this.handleToggleAdvanced = this.handleToggleAdvanced.bind(this)
    this.handleNextClick = this.handleNextClick.bind(this)
    this.handlePreviousClick = this.handlePreviousClick.bind(this)
    this.handleSelectResult = this.handleSelectResult.bind(this)
  }

  handleQueryUpdate(query) {
    const { onQueryUpdate } = this.props
    if (onQueryUpdate) onQueryUpdate(query)
  }

  handleToggleAdvanced() {
    const { onToggleAdvanced } = this.props
    if (onToggleAdvanced) onToggleAdvanced(false)
  }

  handleNextClick() {
    const { onNextClick } = this.props
    if (onNextClick) onNextClick()
  }

  handlePreviousClick() {
    const { onPreviousClick } = this.props
    if (onPreviousClick) onPreviousClick()
  }

  handleSelectResult() {
    const { onSelectResult } = this.props
    this.setState(() => ({ quickSearchOpen: false }))
    if (onSelectResult) onSelectResult()
  }

  render() {
    const {
      routedQuery,
      currentQuery,
      bundle,
      results,
      loading,
      fhirServer,
    } = this.props
    // If the query has been updated within state, use that over props.
    const query =
      currentQuery === null || currentQuery === undefined
        ? routedQuery
        : currentQuery
    return (
      <div className='search-advanced'>
        <div className='search-advanced-form'>
          <TextField
            value={query}
            placeholder='Search'
            className='search-input'
            disabled
            onChange={this.handleQueryUpdate}
          />
          <SearchForm
            fhirServer={fhirServer}
            query={query}
            onSearchUpdate={this.handleQueryUpdate}
          />
          <Loading loading={loading}>
            <Expand
              active
              className='search-toggle-advanced'
              onToggle={this.handleToggleAdvanced}
            />
          </Loading>
        </div>
        {results ? (
          <div className='search-advanced-results'>
            <SearchSummary
              totalResults={bundle.total}
              nextLink={nextLinkFromBundle(bundle)}
              previousLink={previousLinkFromBundle(bundle)}
              onNextClick={this.handleNextClick}
              onPreviousClick={this.handlePreviousClick}
            />
            <FullSearchResults
              query={query}
              results={results}
              onSelectResult={this.handleSelectResult}
            />
          </div>
        ) : null}
      </div>
    )
  }
}

export default AdvancedSearch
