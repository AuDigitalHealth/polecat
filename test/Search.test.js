import React from 'react'
import { shallow } from 'enzyme'
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import sinon from 'sinon'

import { Search } from '../src/Search.js'
import AdvancedSearch from '../src/AdvancedSearch.js'
import config from './config.js'

import searchBundle1 from './fixtures/searchBundle-1.json'
import searchBundle2 from './fixtures/searchBundle-2.json'

var mock = new MockAdapter(axios)

describe('Search', () => {
  it('should retrieve all results on download click', () => {
    const props = {
      fhirServer: config.fhirServer,
      history: {},
    }
    const wrapper = shallow(<Search {...props} />)
    wrapper.setState({
      query: 'type:MP ancestor:37732011000036107|phenylephrine',
      advanced: true,
    })
    // Mock out HTTP calls such that the search request receives the first page,
    // and the subsequent request receives the second page of results.
    mock
      .onGet(
        `${
          props.fhirServer
        }/Medication?medication-resource-type=UPD&ancestor=Medication/37732011000036107&status=active,inactive,entered-in-error&_summary=true&_count=100`,
      )
      .reply(200, searchBundle1, { 'content-type': 'application/fhir+json' })
      .onGet(
        `${
          props.fhirServer
        }?_getpages=ac0f1c84-093d-4696-af6a-d8245e92f846&_getpagesoffset=20&_count=20&_bundletype=searchset`,
      )
      .reply(200, searchBundle2, { 'content-type': 'application/fhir+json' })
    // Get the `onDownloadClick` handler passed to the AdvancedSearchComponent,
    // and call it.
    const onDownloadClick = wrapper.find(AdvancedSearch).prop('onDownloadClick')
    onDownloadClick()
    // Give the async code a chance to execute, then check that the `allResults`
    // prop has been passed to the AdvancedSearch component, with the correct value.
    return new Promise(resolve => {
      setTimeout(() => {
        wrapper.update()
        const advancedSearch = wrapper.find(AdvancedSearch)
        expect(advancedSearch).toBeDefined()
        expect(advancedSearch.prop('allResults')).toBeDefined()
        expect(advancedSearch.prop('allResults')).toMatchSnapshot()
        resolve()
      }, 50)
    })
  })

  it('should download with query in props but not state', () => {
    const props = {
      fhirServer: config.fhirServer,
      query: 'type:MP ancestor:37732011000036107|phenylephrine',
      history: {},
    }
    const wrapper = shallow(<Search {...props} />)
    // Mock out HTTP calls such that the search request receives the first page,
    // and the subsequent request receives the second page of results.
    mock
      .onGet(
        `${
          props.fhirServer
        }/Medication?medication-resource-type=UPD&ancestor=Medication/37732011000036107&status=active,inactive,entered-in-error&_summary=true&_count=100`,
      )
      .reply(200, searchBundle1, { 'content-type': 'application/fhir+json' })
      .onGet(
        `${
          props.fhirServer
        }?_getpages=ac0f1c84-093d-4696-af6a-d8245e92f846&_getpagesoffset=20&_count=20&_bundletype=searchset`,
      )
      .reply(200, searchBundle2, { 'content-type': 'application/fhir+json' })
    // Get the `onDownloadClick` handler passed to the AdvancedSearchComponent,
    // and call it.
    const onDownloadClick = wrapper.find(AdvancedSearch).prop('onDownloadClick')
    onDownloadClick()
    // Give the async code a chance to execute, then check that the `allResults`
    // prop has been passed to the AdvancedSearch component, with the correct value.
    return new Promise(resolve => {
      setTimeout(() => {
        wrapper.update()
        const advancedSearch = wrapper.find(AdvancedSearch)
        expect(advancedSearch).toBeDefined()
        expect(advancedSearch.prop('allResults')).toBeDefined()
        resolve()
      }, 50)
    })
  })

  describe('request triggering', () => {
    let httpSpy
    beforeEach(() => (httpSpy = sinon.spy(axios, 'get')))
    afterEach(() => httpSpy.restore())

    it('should not make another request if sent the same query through props', () => {
      const props = {
        fhirServer: config.fhirServer,
        history: {},
        query: 'g',
      }
      mock
        .onGet(
          `${
            props.fhirServer
          }/Medication?status=active,inactive,entered-in-error&_text=g&_summary=true&_count=100`,
        )
        .reply(200, searchBundle1, {
          'content-type': 'application/fhir+json',
        })
      const wrapper = shallow(<Search {...props} />)
      return new Promise(resolve => {
        setTimeout(() => {
          wrapper.setProps({ loading: true, query: 'g' })
          expect(httpSpy.callCount).toBe(1)
          resolve()
        }, 50)
      })
    })

    it('should not make another request if sent the same query through onQueryUpdate', () => {
      const props = {
        fhirServer: config.fhirServer,
        history: { push: jest.fn() },
        query: 'g',
      }
      mock
        .onGet(
          `${
            props.fhirServer
          }/Medication?status=active,inactive,entered-in-error&_text=g&_summary=true&_count=100`,
        )
        .reply(200, searchBundle1, {
          'content-type': 'application/fhir+json',
        })
      const wrapper = shallow(<Search {...props} />)
      const onQueryUpdate = wrapper.find(AdvancedSearch).prop('onQueryUpdate')
      return new Promise(resolve => {
        setTimeout(() => {
          onQueryUpdate('g')
          expect(httpSpy.callCount).toBe(1)
          resolve()
        }, 50)
      })
    })

    it('should not make another request if the same query comes through onQueryUpdate, then props', () => {
      const props = {
        fhirServer: config.fhirServer,
        history: { push: jest.fn() },
      }
      mock
        .onGet(
          `${
            props.fhirServer
          }/Medication?status=active,inactive,entered-in-error&_text=g&_summary=true&_count=100`,
        )
        .reply(200, searchBundle1, {
          'content-type': 'application/fhir+json',
        })
      const wrapper = shallow(<Search {...props} />)
      wrapper.setState({ advanced: true })
      const onQueryUpdate = wrapper.find(AdvancedSearch).prop('onQueryUpdate')
      onQueryUpdate('g')
      return new Promise(resolve => {
        setTimeout(() => {
          wrapper.setProps({ query: 'g' })
          expect(httpSpy.callCount).toBe(1)
          resolve()
        }, 50)
      })
    })
  })
})
