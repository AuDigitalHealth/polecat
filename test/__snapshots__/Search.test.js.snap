// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should retrieve all results on download click 1`] = `
Array [
  Object {
    "coding": Array [
      Object {
        "code": "61786011000036109",
        "display": "chlorphenamine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61803011000036104",
        "display": "dextromethorphan + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2014-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61800011000036107",
        "display": "phenylephrine + paracetamol",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-01-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61764011000036104",
        "display": "paracetamol + phenylephrine + dextromethorphan",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-04-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "21375011000036101",
        "display": "prednisolone acetate + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2015-01-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61726011000036103",
        "display": "brompheniramine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2014-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61722011000036102",
        "display": "brompheniramine + phenylephrine + dextromethorphan",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61802011000036106",
        "display": "phenylephrine + paracetamol + chlorphenamine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61725011000036105",
        "display": "paracetamol + codeine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-01-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61724011000036106",
        "display": "guaifenesin + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61754011000036107",
        "display": "paracetamol + phenylephrine + guaifenesin",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-05-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61812011000036109",
        "display": "diphenhydramine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2014-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "74988011000036106",
        "display": "lidocaine (lignocaine) + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-12-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "926976011000036103",
        "display": "phenylephrine + pholcodine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2014-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "879421000168105",
        "display": "zinc sulfate + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-02-28",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61788011000036108",
        "display": "pholcodine + phenylephrine + chlorphenamine + ammonium chloride",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "85071011000036108",
        "display": "phenylephrine + ibuprofen",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-06-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "1067911000168106",
        "display": "paracetamol + codeine + phenylephrine + chlorphenamine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2018-01-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "776541000168109",
        "display": "bromhexine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-04-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "69881011000036108",
        "display": "phenylephrine + chlorhexidine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-03-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61815011000036101",
        "display": "paracetamol + dextromethorphan + phenylephrine + chlorphenamine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2017-04-30",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61756011000036104",
        "display": "ascorbic acid + aspirin + chlorphenamine + phenylephrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
  Object {
    "coding": Array [
      Object {
        "code": "61814011000036100",
        "display": "ammonium chloride + codeine + guaifenesin + phenylephrine + pseudoephedrine",
        "system": "http://snomed.info/sct",
      },
    ],
    "lastModified": "2016-10-31",
    "sourceCodeSystemUri": "http://snomed.info/sct",
    "sourceCodeSystemVersion": "http://snomed.info/sct?version=http%3A%2F%2Fsnomed.info%2Fsct%2F32506021000036107%2Fversion%2F20180228",
    "status": "active",
    "subsidy": Array [],
    "type": "MP",
  },
]
`;
